File: dto/dtos/Country.cs
using dto.Paging;
using System.ComponentModel.DataAnnotations;

namespace dto.dtos
{
    public class CountryRes
    {
        public int CountryId { get; set; }
        public string? Slug { get; set; }

        // Child tables
        public ICollection<StateRes>? States { get; set; }
        public IEnumerable<CountryNameRes>? CountryNames { get; set; }
    }
    public class CountryReqEdit
    {
        [Required]
        public string? Slug { get; set; }
    }
    public class CountryReqSearch : PagedReq
    {

    }
}

File: admin/src/dtos/Country.ts
import Common from "../utility/Common";
import { CountryNameRes } from "./CountryName";
import { PagedReq } from "./PagedReq";
import { StateRes } from "./State";

export interface CountryRes {
  countryId?: string;
  slug?: string;
  states?: StateRes[];
  countryNames?: CountryNameRes[];
}

export class CountryReqEdit {
  slug?: string = "";
}

export class CountryReqSearch extends PagedReq {
  constructor(
    {
      pageNumber = 1,
      pageSize = Common.DEFAULT_PAGE_SIZE,
      orderBy,
      searchText = "",
    }: PagedReq,
    {}
  ) {
    super({
      pageNumber: pageNumber,
      pageSize: pageSize,
      orderBy: orderBy,
      searchText: searchText,
    });
  }
}
